### PR TITLE
feat(sentry): use @sentry/react

### LINKED ISSUE #952: Improve client-side monitoring with @sentry/react
Currently the codebase uses @sentry/browser. Upgrade to [@sentry/react](https://docs.sentry.io/platforms/javascript/guides/react/) instead; expected benefits include performance monitoring and [integrations](https://docs.sentry.io/platforms/javascript/guides/react/configuration/integrations/) with Redux and React Router.

### PR DESCRIPTION
**Pull request by LoneRifle**
## Problem

Closes #952 

## Solution

Drop @sentry/react in to replace @sentry/browser

**New dependencies**:

- `@sentry/react` : 1-for-1 replacement for `@sentry/browser`

### CODE REVIEW

**Review by liangyuanruo — CHANGES_REQUESTED** [review]
Put in the integration for [Redux](https://docs.sentry.io/platforms/javascript/guides/react/configuration/integrations/redux/) and  [React Router](https://docs.sentry.io/platforms/javascript/guides/react/configuration/integrations/react-router/)?

File: src/client/app/components/pages/RootPage/index.tsx [hunk 1]
```diff
@@ -31,12 +34,13 @@ import theme from '../../../theme'
...
   store: Store
+  history: History
 }
 
-const Root: FunctionComponent<RootProps> = ({ store }: RootProps) => (
+const Root: FunctionComponent<RootProps> = ({ store, history }: RootProps) => (
   <Provider store={store}>
     <MuiThemeProvider theme={theme}>
-      <HashRouter>
+      <Router history={history}>
```
Comment (liangyuanruo) [inline src/client/app/components/pages/RootPage/index.tsx:43]: are you confident that this change doesn't break anything? what does the new route look like? will it clash with the redirection endpoint?

Comment (LoneRifle) [reply src/client/app/components/pages/RootPage/index.tsx:43]: New route looks exactly the same - using `<HashRouter>` and using `<Router>` with `createHashHistory()` are equivalent.

Comment (liangyuanruo) [reply src/client/app/components/pages/RootPage/index.tsx:43]: thanks!

**Review by liangyuanruo — APPROVED** [review]
lgtm!

**Pull request merged**